fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGeneratorBase.java

```diff
@@ -85,6 +86,16 @@ protected ListIterator<Symbol> actionIterator(FieldAction action) {
     while (actionIterator.hasNext()) {
       Symbol symbol = actionIterator.next();
 
+      if (Symbol.Kind.REPEATER.equals(symbol.kind)
+          && "array-end"
+          .equals(
+              getSymbolPrintName(
```

Comment:
this method is fairly slow, and this class (Symbol) is back ported - so we can change it.

is it worth making getPrintName() public so it could be accessed here?

Reply:
Terminal symbol toString returns the print name.  Terminal is the only Symbol with printName.  Re-worked to use Terminal method toString to get printName